### PR TITLE
Rebuild currency service Dockerfile with alpine

### LINKED ISSUE #681: Optimize c++ currency service build time
# Feature Request

The currency service build time is drastically slowing our release process and a general pain. 

Please help.

### PR DESCRIPTION
**Pull request by svrnm**
# Changes

This should fix #681, #679 partially and #574: Alpine linux provides a [grpc-dev package](https://pkgs.alpinelinux.org/package/edge/community/x86/grpc-dev) that can be used to skip all the dependencies for building opentelemetry-cpp

This brings down the build time to **2m 30s** on an intel macbook pro (2015)

## Merge Requirements

* [ ] `CHANGELOG.md` updated to document new feature additions
* [ ] Appropriate documentation updates in the [docs](../docs/) folder

### CODE REVIEW

**Comment by cartersocha** [comment]
This made a HUGE difference on my machine. 43 vs.  407 seconds on my m2 mac book air. Thank you so much @svrnm 

@austinlparker we should release a new version immediately after we merge this and test the new build performance. Then maybe we can add another quick release for arm64 support unless other services continue to block us. 

I'd be curious if we could do something similar for the other slow services? fyi @GaryPWhite

**Review by cartersocha — APPROVED** [review]

**Comment by austinlparker** [comment]
Wouldn't this also work for the rust image since its the same libs? Worth a shot.

File: src/currencyservice/src/server.cpp [hunk 1]
```diff
@@ -176,10 +176,10 @@ class CurrencyService final : public hipstershop::CurrencyService::Service
     std::string span_name = "CurrencyService/Convert";
     auto span =
         get_tracer("currencyservice")->StartSpan(span_name,
```
Comment (julianocosta89) [inline src/currencyservice/src/server.cpp:178]: Could you also update this in here?
https://github.com/open-telemetry/opentelemetry-demo/blob/main/docs/services/currencyservice.md#create-new-spans

**Review by julianocosta89 — CHANGES_REQUESTED** [review]

**Review by julianocosta89 — APPROVED** [review]

**Review by lalitb — APPROVED** [review]
Nicely done. Thanks.

File: src/currencyservice/src/server.cpp (hunk 1 above)
Comment (svrnm) [reply src/currencyservice/src/server.cpp:178]: ah yes, thx, done!

**Comment by svrnm** [comment]
Additional update: To reduce the final image size by ~300 MB (down to 65 on arm64) I turned it into a multi-stage build.

**Review by fatsheep9146 — APPROVED** [review]

**Comment by mviitane** [comment]
Thanks @svrnm, this was an amazing improvement!

**Pull request merged**